crates/matrix-sdk/tests/integration/event_cache.rs

```diff
@@ -146,46 +149,63 @@ async fn test_add_initial_events() {
 #[async_test]
 async fn test_ignored_unignored() {
     let server = MatrixMockServer::new().await;
-    let client = server.client_builder().build().await;
+
+    let event_cache_store = Arc::new(MemoryStore::new());
+    let client = server
+        .client_builder()
+        .store_config(
+            StoreConfig::new("hodor".to_owned()).event_cache_store(event_cache_store.clone()),
+        )
+        .build()
+        .await;
 
     // Immediately subscribe the event cache to sync updates.
     client.event_cache().subscribe().unwrap();
+    client.event_cache().enable_storage().unwrap();
 
-    // If I sync and get informed I've joined The Room, but with no events,
     let room_id = room_id!("!omelette:fromage.fr");
     let other_room_id = room_id!("!galette:saucisse.bzh");
 
-    server.sync_joined_room(&client, room_id).await;
-    server.sync_joined_room(&client, other_room_id).await;
-
     let dexter = user_id!("@dexter:lab.org");
     let ivan = user_id!("@ivan:lab.ch");
     let f = EventFactory::new();
 
-    // If I add initial events to a few rooms,
-    client
-        .event_cache()
-        .add_initial_events(
+    // Given two rooms which add initial items,
+    event_cache_store
+        .handle_linked_chunk_updates(
             room_id,
             vec![
-                f.text_msg("hey there").sender(dexter).into_sync(),
-                f.text_msg("hoy!").sender(ivan).into_sync(),
+                Update::NewItemsChunk { previous: None, new: ChunkIdentifier::new(0), next: None },
+                Update::PushItems {
+                    at: Position::new(ChunkIdentifier::new(0), 0),
+                    items: vec![
+                        f.text_msg("hey there").sender(dexter).into_sync(),
+                        f.text_msg("hoy!").sender(ivan).into_sync(),
+                    ],
+                },
```

Comment:
I wonder if we shouldn't use a builder for that 🤔, thoughts?

Reply:
I think we could even avoid that by using a sync, and let the event cache store the events in memory.